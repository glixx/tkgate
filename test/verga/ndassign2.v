//
// Non-blocking assignment test.
//
module top;
  reg[31:0] a,b,c;
  integer i;

  initial
      begin
	#1 a = 1;
	#1 b = 2;
	$display("%d: a=%d b=%d",$time,a,b);
	a <= b;
	b <= a;
	#1 $display("%d: a=%d b=%d",$time,a,b);
	a <= #2 b;
	b <= #3 a;
	#1 $display("%d: a=%d b=%d",$time,a,b);
	#1 $display("%d: a=%d b=%d",$time,a,b);
	#1 $display("%d: a=%d b=%d",$time,a,b);
	#1 $display("%d: a=%d b=%d",$time,a,b);
	$display("");
        c = 1;
        c <= @(a) 2;
        a <= #2 3;
	#1 $display("%d: a=%d c=%d",$time,a,c);
	@(c) $display("%d: a=%d c=%d",$time,a,c);
	#1 $display("%d: a=%d c=%d",$time,a,c);
      end 

endmodule

