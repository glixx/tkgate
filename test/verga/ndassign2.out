2: a=1 b=2
3: a=2 b=1
4: a=2 b=1
5: a=2 b=1
6: a=1 b=1
7: a=1 b=2

8: a=1 c=1
9: a=3 c=2
10: a=3 c=2
